TulparAudio/include/tulpar/TulparConfigurator.hpp

```diff
@@ -0,0 +1,62 @@
+/*
+* Copyright (C) 2018 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef TULPAR_TULPAR_CONFIGURATOR_HPP
+#define TULPAR_TULPAR_CONFIGURATOR_HPP
+
+#include <cstdint>
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace tulpar
+{
+
+/** @brief  Tulpar configuration object */
+class TulparConfigurator
+{
+public:
+    /** @brief  Device description */
+    struct Device
+    {
+        //! Device name
+        std::string name;
+
+        //! Flag indicating if device is default
+        bool isDefault;
+    };
+
+    //! Creates configuration object
+    TulparConfigurator();
```

Comment:
I think good idea is to use explicit default initialization of data members during the declaration. This way you could avoid writing a non-default constructor and explicitly define default destruction and etc.

Reply:
I did not fully get what you want to change. Could you give an example?